src/pcre2_compile.h

```diff
@@ -218,11 +218,26 @@ therefore no need for it to have a length entry, so use a high value. */
 #define CLIST_ALIGN_TO(base, align) \
   ((base + ((size_t)(align) - 1)) & ~((size_t)(align) - 1))
 
+/* Structure for holding information about an OP_ECLASS internal operand.
+An "operand" here could be just a single OP_[X]CLASS, or it could be some
+complex expression; but it's some sequence of ECL_* codes which pushes one
+value to the stack. */
+typedef struct {
+  /* The position of the operand - or NULL if (lengthptr != NULL). */
+  PCRE2_UCHAR *code_start;
+  PCRE2_SIZE length;
+  /* The operand's type if it is a single code (ECL_XCLASS, ECL_ANY, ECL_NONE);
+  otherwise zero if the operand is not atomic. */
+  uint8_t op_single_type;
+  /* Regardless of whether it's a single code or not, we fully constant-fold
+  the bitmap for code points < 256. */
+  uint8_t bits[32];
```

Comment:
I mentioned this could be an uint32_t array as well using an union, and merging / counting 32 bit values are 4 times faster than bytes.